crates/transaction-pool/src/pool/pending.rs

```diff
@@ -154,6 +155,35 @@ impl<T: TransactionOrdering> PendingPool<T> {
         best
     }
 
+    /// Same as `best` but also includes the given unlocked transactions and also apply new
+    /// attributes.
+    pub(crate) fn best_with_unlocked_and_attributes(
+        &self,
+        unlocked: Vec<Arc<ValidPoolTransaction<T::Transaction>>>,
+        origin_base_fee: u64,
+        attributes: BestTransactionsAttributes,
```

Comment:
this looks strange, we shouldn't need this

this function should be the same as best_with_basefee_and_blobfee but with an additional unlocked vec